packages/phishing-controller/src/utils.test.ts

```diff
@@ -140,3 +148,110 @@ describe('applyDiffs', () => {
     });
   });
 });
+
+describe('validateConfig', () => {
+  it('throws an error if the config is not an object', () => {
+    expect(() => validateConfig(null)).toThrow('Invalid config');
+  });
+
+  it('throws an error if the config contains a tolerance without a fuzzylist', () => {
+    expect(() => validateConfig({ tolerance: 2 })).toThrow(
+      'Fuzzylist tolerance provided without fuzzylist',
+    );
+  });
+
+  it('throws an error if the config contains an invalid name', () => {
+    expect(() => validateConfig({ name: 123 })).toThrow(
+      "Invalid config parameter: 'name'",
+    );
+  });
+
+  it('throws an error if the config contains an invalid version', () => {
+    expect(() => validateConfig({ version: { foo: 'bar' } })).toThrow(
+      "Invalid config parameter: 'version'",
+    );
+  });
+});
+
+describe('domainToParts', () => {
+  it('correctly converts a domain string to an array of parts', () => {
+    const domain = 'example.com';
+    const result = domainToParts(domain);
+    expect(result).toStrictEqual(['com', 'example']);
+  });
+
+  it('correctly converts a domain string with subdomains to an array of parts', () => {
+    const domain = 'sub.example.com';
+    const result = domainToParts(domain);
+    expect(result).toStrictEqual(['com', 'example', 'sub']);
+  });
+
+  it('throws an error if the domain string is invalid', () => {
+    // @ts-expect-error testing invalid input
+    expect(() => domainToParts(123)).toThrow('123');
+  });
+});
+
+describe('processConfigs', () => {
+  it('correctly converts a list of configs to a list of processed configs', () => {
+    const configs = [
+      {
+        allowlist: ['example.com'],
+        blocklist: ['sub.example.com'],
+        fuzzylist: ['fuzzy.example.com'],
+        tolerance: 2,
+      },
```

Comment:
What are your thoughts on testing with multiple configs rather than just one?

Reply:
More sounds better but also extending with further tests maybe better in follow-up as this ports existing  code from an already utilized package?

Reply:
I'm also more inclined to leave improvements over the original code / tests for follow-up PRs, but definitely something we should do

Reply:
Okay no problem.